src/python/pants/option/options_test.py

```diff
@@ -1733,3 +1748,82 @@ class DummyOptionable1(Optionable):
         single_warning_dummy1 = assert_single_element(w)
         self.assertEqual(single_warning_dummy1.category, DeprecationWarning)
         self.assertEqual("vv", vals1.foo)
+
+    def test_list_of_enum_single_value(self) -> None:
+        options = self._parse(flags="other-enum-scope --some-list-enum=another-value")
+        assert [self.SomeEnumOption.another_value] == options.for_scope(
+            "other-enum-scope"
+        ).some_list_enum
+
+    def test_list_of_enum_default_value(self) -> None:
+        options = self._parse(flags="other-enum-scope --some-list-enum-with-default=another-value")
+        assert [
+            self.SomeEnumOption.yet_another,
+            self.SomeEnumOption.another_value,
+        ] == options.for_scope("other-enum-scope").some_list_enum_with_default
+        options = self._parse()
+        assert [self.SomeEnumOption.yet_another] == options.for_scope(
+            "other-enum-scope"
+        ).some_list_enum_with_default
+
+    def test_list_of_enum_multiple_values(self) -> None:
+        options_1 = self._parse(flags="other-enum-scope --some-list-enum=another-value,one-more")
+        options_2 = self._parse(
+            flags="other-enum-scope --some-list-enum=yet-another,one-more,a-value"
+        )
+
+        assert [
+            self.SomeEnumOption.another_value,
+            self.SomeEnumOption.one_more,
+        ] == options_1.for_scope("other-enum-scope").some_list_enum
+
+        assert [
+            self.SomeEnumOption.yet_another,
+            self.SomeEnumOption.one_more,
+            self.SomeEnumOption.a_value,
+        ] == options_2.for_scope("other-enum-scope").some_list_enum
+
+    def test_list_of_enum_from_config(self) -> None:
+        options = self._parse(
+            config={"other-enum-scope": {"some_list_enum": "['one-more','a-value']"}}
```

Comment:
Nit: space will make it clearer these are distinct
```suggestion
            config={"other-enum-scope": {"some_list_enum": "['one-more', 'a-value']"}}
```